### ISSUE TITLE
Bug: extract_G_loc crashes when Sigma_imp is not set even for with_Sigma=False

### ISSUE THREAD

**Issue by merkelm** [TRIQS/dft_tools#233]
### Description

When executing SumkDFT.extract_G_loc, it always uses `self.Sigma_imp` to initialize the Green's function object G_loc. Therefore, if `self.Sigma_imp` has not been set, it crashes. This probably should not happen when using `with_Sigma=False`, now that SumkDFT has its own mesh object.

The error message is the following:

```
File "/triqs/lib/python3.10/site-packages/triqs_dft_tools/sumk_dft.py", line 761, in extract_G_loc
    G_loc = [self.Sigma_imp[icrsh].copy() for icrsh in range(
  File "/triqs/lib/python3.10/site-packages/triqs_dft_tools/sumk_dft.py", line 761, in <listcomp>
    G_loc = [self.Sigma_imp[icrsh].copy() for icrsh in range(
AttributeError: 'SumkDFT' object has no attribute 'Sigma_imp'
```

### Versions

This happens on the branch `unstable` and is probably not relevant for 3.1.x. I run on commit a072a54.

**Comment by the-hampel** [comment]
Thank you @merkelm for reporting this! PR #234 should fix this. @merkelm could you take a look if this makes sense and works for your case? The code is now simpler. I hope this still covers all use cases. Tests are passing here and in solid_dmft. If you agree I will merge it.

**Comment by merkelm** [comment]
The code changes look good and it worked in the one test I did. Thanks for fixing this!

**Comment by the-hampel** [comment]
fixed merging #234
